app/services/katello/pulp3/smart_proxy_repository.rb

```diff
@@ -33,6 +33,38 @@ def delete_orphaned_repositories_for_mirror_proxies
           ::Katello::Pulp3::Repository.new(nil, smart_proxy).repositories_api.delete(repo._href)
         end
       end
+
+      def pulp3_enabled_repo_types(content_types = SETTINGS[:katello][:content_types].keys)
+        repo_types = content_types.collect do |content_type|
+          Katello::RepositoryTypeManager.find(content_type)
+        end
+
+        repo_types.select { |type| type.try(:pulp3_service_class) }
+      end
+
+      def delete_orphaned_distributions_for_mirror_proxies(repo_types = nil)
+        repo_types ||= pulp3_enabled_repo_types
```

Comment:
could you just make pulp3_enabled_repo_types be the default value for  'repo_types' when passed in?